8vim/src/main/java/inc/flide/vim8/ui/LauncherActivity.java

```diff
@@ -72,17 +72,22 @@ public void askUserPreferredEmoticonKeyboard(){
         }
         ArrayList<String> keyboardIds = new ArrayList<>(inputMethodsNameAndId.values());
 
+        SharedPreferences sp = getSharedPreferences(getString(R.string.basic_preference_file_name), Activity.MODE_PRIVATE);
+        String selectedId = sp.getString("selected_emoticon_keyboard","");
```

Comment:
within the getString function try not using the literal string. Please change it to R.string.bp_selected_emoticon_keyboard

Reply:
Also, the default value should be '-1' and not empty (""), in case the selected keyboard gets disabled or uninstalled in the interim. Also, your code already handles the -1 scenario, so there will be nothing to worry about.